internal/dataplane/kong_client.go

```diff
@@ -439,6 +456,35 @@ func (c *KongClient) Update(ctx context.Context) error {
 	return nil
 }
 
+func (c *KongClient) FetchLastGoodConfiguration(ctx context.Context) error {
+	gatewayClients := c.clientsProvider.GatewayClients()
+	c.logger.Debugf("fetching last good configuration from %d gateway clients", len(gatewayClients))
+
+	var goodKongState *kongstate.KongState
+	var errs error
+	for _, client := range gatewayClients {
+		rs, err := c.kongConfigFetcher.GetKongRawState(ctx, client.AdminAPIClient())
```

Comment:
How do we decide if we got different configurations from multiple gateways? Seems that we use the first non-empty config.

Reply:
This is actually a good point. My first intention was to compare all of them and get the config only if there were equal. Then I pursued the way of getting the first one and storing it as the last good one. This way, we can restore the expected status of all the proxies with the same configuration. I think that having different proxies with different loaded configurations is something that should never happen (and if it does, it's because of strange behaviors or events). @mheap do you have some thoughts on how this should behave?

Reply:
We can safely assume that all nodes contain the same configuration. If that's not the case, I expect it to be a bug. If it's something intentional, we can revisit in the future.